src/libpriv/rpmostree-core.c

```diff
@@ -3874,6 +3912,9 @@ rpmostree_context_assemble (RpmOstreeContext      *self,
       if (pkg == filesystem_package)
         continue;
 
+      if (rpmte_is_kernel (te))
+        self->kernel_changed = TRUE;
```

Comment:
Yeah, I was just being conservative.  I can't offhand think of a situation where we're installing a kernel (and not removing one) *and* a higher level would want to react to it in some way, but it doesn't hurt to make the flag correct right?

Reply:
> I can't offhand think of a situation where we're installing a kernel (and not removing one)

We forbid this anyway right? In `prepare`, I see `dnf_sack_set_installonly_limit`. And indeed trying out a `rpm-ostree install ./kernel-*.rpm` triggers the `Forbidden base package replacements:` error.

> but it doesn't hurt to make the flag correct right?

Yeah, definitely minor :). It's more that having it there implies it could be meaningful. It's like my preference for `g_str_equal` vs `g_strcmp0` when possible.

Reply:
Yeah, but think about the buildroot case where the kernel may end up being pulled in as a dependency.

Reply:
Ahh yep, good point.